record-and-playback/core/lib/recordandplayback/edl/video.rb

```
@@ -312,6 +315,55 @@ def self.render(edl, layout, output_basename, threads = 2)
+          # Ensure videofile is not already a resampled one from a previous remux step if names overlap
+          next if videofile.include?("_resampled.") || videofile.include?(".nut")
+
+          gap_details = BigBlueButton::EDL::MediaUtils.has_large_pts_gaps?(videofile, MAX_GAP_FOR_RESAMPLE_MS, 'v')
+          if gap_details
+            BigBlueButton.logger.info "Video '#{File.basename(videofile)}' identified with large PTS gap between #{gap_details[0].round(3)}s and #{gap_details[1].round(3)}s. Marking for segmented resampling."
+            videos_to_resample_due_to_gaps[videofile] = gap_details
+          end
+        end
+
+        if videos_to_resample_due_to_gaps.any?
+          BigBlueButton.logger.info "Resampling #{videos_to_resample_due_to_gaps.length} video(s) with large PTS gaps (segmented approach)."
```

RuboCop (1.82.1)

[convention] Line is too long. [140/128]

(Layout/LineLength)